include/RAJA/policy/cuda/intrinsics.hpp

```diff
@@ -415,18 +415,18 @@ RAJA_DEVICE RAJA_INLINE T block_reduce(T val, T identity)
   }
 
   // reduce per warp values
-  if (numThreads > policy::cuda::WARP_SIZE) {
+  if (numThreads > policy::cuda::device_constants.WARP_SIZE) {
 
-    static_assert(policy::cuda::MAX_WARPS <= policy::cuda::WARP_SIZE,
+    static_assert(policy::cuda::device_constants.MAX_WARPS <= policy::cuda::device_constants.WARP_SIZE,
```

Comment:
I don't understand this constraint. To me, it seems that ``MAX_WARPS`` and ``WARP_SIZE`` are independent, unless ``MAX_WARPS`` doesn't mean what I think it does.

Reply:
This is true, but we assume this in the reducer implementation, so I suppose that this assert should be there instead.

Reply:
That makes more sense, I think

Reply:
Updated the message to be more along the lines of my message.